lib/runtime/wasmer/imports.go

```diff
@@ -185,45 +188,191 @@ func ext_sandbox_memory_teardown_version_1(context unsafe.Pointer, a C.int32_t)
 }
 
 //export ext_crypto_ed25519_generate_version_1
-func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, a C.int32_t, z C.int64_t) C.int32_t {
+func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, keyTypeID C.int32_t, seedSpan C.int64_t) C.int32_t {
 	logger.Trace("[ext_crypto_ed25519_generate_version_1] executing...")
-	logger.Warn("[ext_crypto_ed25519_generate_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+
+	// TODO: key types not yet implemented
+	// id := memory[idData:idData+4]
+
+	seedBytes := asMemorySlice(instanceContext, seedSpan)
+	buf := &bytes.Buffer{}
+	buf.Write(seedBytes)
+
+	seed, err := optional.NewBytes(false, nil).Decode(buf)
+	if err != nil {
+		logger.Warn("[ext_crypto_ed25519_generate_version_1] cannot generate key", "error", err)
+		return 0
+	}
+
+	var kp crypto.Keypair
+
+	if seed.Exists() {
+		kp, err = ed25519.NewKeypairFromSeed(seedBytes)
+	} else {
+		kp, err = ed25519.GenerateKeypair()
+	}
+
+	if err != nil {
+		logger.Warn("[ext_crypto_ed25519_generate_version_1] cannot generate key", "error", err)
+		return 0
+	}
+
+	runtimeCtx.Keystore.Insert(kp)
+
+	ret, err := toWasmMemorySized(instanceContext, kp.Public().Encode(), 32)
+	if err != nil {
+		logger.Warn("[ext_crypto_ed25519_generate_version_1] failed to allocate memory", "error", err)
+		return 0
+	}
+
+	logger.Debug("[ext_crypto_ed25519_generate_version_1] generated ed25519 keypair", "public", kp.Public().Hex())
+	return C.int32_t(ret)
 }
 
 //export ext_crypto_ed25519_public_keys_version_1
-func ext_crypto_ed25519_public_keys_version_1(context unsafe.Pointer, a C.int32_t) C.int64_t {
+func ext_crypto_ed25519_public_keys_version_1(context unsafe.Pointer, keyTypeID C.int32_t) C.int64_t {
 	logger.Trace("[ext_crypto_ed25519_public_keys_version_1] executing...")
-	logger.Warn("[ext_crypto_ed25519_public_keys_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+	memory := instanceContext.Memory().Data()
+
+	keys := runtimeCtx.Keystore.Ed25519PublicKeys()
+
+	ptr, err := runtimeCtx.Allocator.Allocate(uint32(len(keys) * 32))
+	if err != nil {
+		logger.Error("[ext_crypto_ed25519_public_keys_version_1]", "error", err)
+		return 0
+	}
+
+	for i, key := range keys {
+		copy(memory[ptr+uint32(i*32):ptr+uint32((i+1)*32)], key.Encode())
+	}
```

Comment:
based on the spec, this is expecting a SCALE encoded slice of `[32]byte`, so maybe create a `[][32]byte` with the public keys and encode and return that